frontend/src/metabase/setup/selectors.ts

```diff
@@ -88,39 +90,44 @@ export const getIsEmailConfigured = (state: State): boolean => {
   return getSetting(state, "email-configured?");
 };
 
-export const getSteps = (state: State) => {
-  const usageReason = getUsageReason(state);
-  const activeStep = getStep(state);
-  const tokenFeatures = getSetting(state, "token-features");
-
-  const isPaidPlan =
-    tokenFeatures && Object.values(tokenFeatures).some(value => value === true);
-  const hasAddedPaidPlanInPreviousStep = Boolean(state.setup.licenseToken);
-
-  const shouldShowDBConnectionStep = usageReason !== "embedding";
-  const shouldShowLicenseStep =
-    isEEBuild() && (!isPaidPlan || hasAddedPaidPlanInPreviousStep);
-
-  const steps: { key: SetupStep; isActiveStep: boolean }[] = [
-    { key: "welcome" as const },
-    { key: "language" as const },
-    { key: "user_info" as const },
-    { key: "usage_question" as const },
-    shouldShowDBConnectionStep && {
-      key: "db_connection" as const,
-    },
-    shouldShowLicenseStep && { key: "license_token" as const },
-    { key: "data_usage" as const },
-    { key: "completed" as const },
-  ]
-    .filter(isNotFalsy)
-    .map(({ key }) => ({
-      key,
-      isActiveStep: activeStep === key,
-    }));
-
-  return steps;
-};
+export const getSteps = createSelector(
```

Comment:
Toolkit was complaining that the returned value was not stable, because it was a new array each time, and could have lead to re-renders.